File: 70-climbing-stairs/70-climbing-stairs.py
class Solution:
    dp = [-1]*46
    dp[0]=1
    dp[1]=1
    def findAns(self,n):
        if(self.dp[n]!=-1):
            return self.dp[n]
        elif(n<0):
            return 0
        else:
            self.dp[n] = self.findAns(n-1) + self.findAns(n-2)
            return self.dp[n]
    def climbStairs(self, n: int) -> int:
        return self.findAns(n)

File: 70-climbing-stairs/70-climbing-stairs.java
class Solution {
    public int result(int n, int []dp){
        if (n==0 || n==1){
            dp[n] = 1;
            return dp[n];
        }
        if(dp[n] != -1){
            return dp[n];
        }
        dp[n] = result(n-1,dp) + result(n-2,dp);
        System.out.println(Arrays.toString(dp));
        return dp[n];
    }
    public int climbStairs(int n) {
        int[] dp = new int[n+1];
        for (int i=0;i<n+1;i++){
            dp[i] = -1;
        }
        return result(n,dp);
    }
}